lib/datadog/statsd.rb

```diff
@@ -337,35 +343,69 @@ def format_event(title, text, opts={})
     private_constant :NEW_LINE, :ESC_NEW_LINE, :COMMA, :BLANK, :PIPE, :DOT,
       :DOUBLE_COLON, :UNDERSCORE
 
-    def escape_event_content(msg)
-      msg.gsub NEW_LINE, ESC_NEW_LINE
+    def escape_event_content!(msg)
+      msg.gsub!(NEW_LINE, ESC_NEW_LINE) || msg
     end
 
-    def escape_tag_content(tag)
-      remove_pipes(tag).gsub COMMA, BLANK
+    def escape_tag_content!(tag)
```

Comment:
We often freeze tags. I believe this will cause an exception to raise for frozen strings. Should we use gsub for frozen strings or something else? 

```
irb(main):001:0> RPC_STORE = "rpc_store:memcached".freeze
=> "memcached"
irb(main):002:0> RPC_STORE.gsub!("|", "")
RuntimeError: can't modify frozen String
	from (irb):2:in `gsub!'
	from (irb):2
	from /Users/jnunemaker/.rbenv/versions/2.2.4/bin/irb:11:in `<main>'
irb(main):003:0> RPC_STORE.gsub("|", "")
=> "rpc_store:memcached"
```